File: code/src/test/java/bits/AddBinaryTest.java
package bits;

import bits.AddBinary.AddBinary;
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class AddBinaryTest {
    AddBinary addBinary = new AddBinary();

    @Test
    void testAddBinary() {
        // Test case 1: Adding two simple binary numbers "1010" + "1011"
        String a1 = "1010";
        String b1 = "1011";
        String expected1 = "10101";
        assertEquals(expected1, addBinary.addBinary(a1, b1));

        // Test case 2: Adding two binary numbers with different lengths "110" + "10"
        String a2 = "110";
        String b2 = "10";
        String expected2 = "1000";
        assertEquals(expected2, addBinary.addBinary(a2, b2));

        // Test case 3: Adding binary numbers where both are "0"
        String a3 = "0";
        String b3 = "0";
        String expected3 = "0";
        assertEquals(expected3, addBinary.addBinary(a3, b3));

        // Test case 4: Adding binary number with itself "1" + "1"
        String a4 = "1";
        String b4 = "1";
        String expected4 = "10";
        assertEquals(expected4, addBinary.addBinary(a4, b4));

        // Test case 5: Adding binary numbers with many carries "1111" + "1"
        String a5 = "1111";
        String b5 = "1";
        String expected5 = "10000";
        assertEquals(expected5, addBinary.addBinary(a5, b5));
    }
}

File: code/src/leetcode/bitManipulation/tests/AddBinaryTest.php
<?php

use PHPUnit\Framework\TestCase;

require_once __DIR__ . '/../AddBinary.php';

class AddBinaryTest extends TestCase
{
  /**
   * Test the addBinary function with various test cases.
   */
  public function testAddBinary()
  {
    // Test case: Adding '1010' and '1011' should return '10101'
    $this->assertEquals('10101', addBinary('1010', '1011'));

    // Test case: Adding '0' and '0' should return '0'
    $this->assertEquals('0', addBinary('0', '0'));

    // Test case: Adding '1' and '1' should return '10'
    $this->assertEquals('10', addBinary('1', '1'));

    // Test case: Adding '111' and '1' should return '1000'
    $this->assertEquals('1000', addBinary('111', '1'));

    // Test case: Adding '1101' and '101' should return '10010'
    $this->assertEquals('10010', addBinary('1101', '101'));
  }
}
